### PR TITLE
Redo font previewer

### PR DESCRIPTION
**Pull request by Sieluna**
https://github.com/AssetRipper/AssetRipper/issues/932 image

### CODE REVIEW

**Comment by ds5678** [comment]
Thank you for the contribution. I can't review it right now, but I'll get to it as soon as I can.

File: Source/AssetRipper.GUI.Web/Pages/Assets/FontTab.cs [hunk 1]
```diff
@@ -7,41 +7,68 @@ internal sealed class FontTab : HtmlTab
...
+				  const fontFace = new FontFace(`{{FileName}}`, `url({{Source}})`);
+				  document.fonts.add(fontFace);
+				  fontFace.load().then().catch(function(error) {
+				    console.error(`Font loading failed: ${error}`);
+				  });
+				  """);
 		}
 	}
+
+	private static string TryDecode(IUnityObjectBase asset, out byte[] data, out string? fileName, out string? mimeType)
```
Comment (ds5678) [inline Source/AssetRipper.GUI.Web/Pages/Assets/FontTab.cs:49]: Methods with this design pattern should always have a boolean return type.

File: Source/AssetRipper.GUI.Web/Pages/Assets/FontTab.cs [hunk 2]
```diff
@@ -7,41 +7,68 @@ internal sealed class FontTab : HtmlTab
...
 	public string? FileName { get; }
 
+	public string? MimeType { get; }
+
 	public override string DisplayName => Localization.AssetTabFont;
 
 	public override string HtmlName => "font";
 
-	public override bool Enabled => Data.Length > 0;
+	public override bool Enabled => !string.IsNullOrEmpty(Source);
```
Comment (ds5678) [inline Source/AssetRipper.GUI.Web/Pages/Assets/FontTab.cs:20]: Why?

**Review by ds5678 — CHANGES_REQUESTED** [review]
Some small changes required, but otherwise good.

File: Source/AssetRipper.GUI.Web/Pages/Assets/FontTab.cs (hunk 2 above)
Comment (Sieluna) [reply Source/AssetRipper.GUI.Web/Pages/Assets/FontTab.cs:20]: I just want it keep the same structure as https://github.com/AssetRipper/AssetRipper/blob/6771232426dc4a38e621ccaf034d916e11a2d79b/Source/AssetRipper.GUI.Web/Pages/Assets/AudioTab.cs#L15

File: Source/AssetRipper.GUI.Web/Pages/Assets/FontTab.cs (hunk 1 above)
Comment (Sieluna) [reply Source/AssetRipper.GUI.Web/Pages/Assets/FontTab.cs:49]: And this part is similar to the code:
https://github.com/AssetRipper/AssetRipper/blob/6771232426dc4a38e621ccaf034d916e11a2d79b/Source/AssetRipper.GUI.Web/Pages/Assets/AudioTab.cs#L39-L46

Comment (Sieluna) [reply Source/AssetRipper.GUI.Web/Pages/Assets/FontTab.cs:49]: Of course, I can put this part of the code like AudioClipDecoder

Comment (ds5678) [reply Source/AssetRipper.GUI.Web/Pages/Assets/FontTab.cs:49]: That code doesn't have any out parameters, which makes it different. In this case, we want a Boolean return. The current return can be made into another out parameter.

File: Source/AssetRipper.GUI.Web/Pages/Assets/FontTab.cs (hunk 2 above)
Comment (ds5678) [reply Source/AssetRipper.GUI.Web/Pages/Assets/FontTab.cs:20]: My intuition says that the previous version is more future-proof. I foresee Source being removed in some way when this becomes asynchronous.

**Pull request merged**